### PR TITLE
Handle RecursionError exception in perform_request (issue 1602)

### PR DESCRIPTION
**Pull request by hmilkovi**
Propagate RecursionError if it happends on lower levels so when used like:
```
import sys
import inspect
import elasticsearch

es_client = elasticsearch.Elasticsearch("http://localhost:9200/")
print(f"Recursion limit: {sys.getrecursionlimit()}")

def rec():
    print(len(inspect.stack()))
    es_client.index(index='testing', body={'test': 'test'}, request_timeout=1)
    rec()

rec()
```
doesn't crash python interpreter with core dump.

Personal opinion is that this is really needed when we you web servers like uwsgi, gunicorn as
then the whole web server dies and doesn't return http 500 status code.

Related issue: https://github.com/elastic/elasticsearch-py/issues/1602

### CODE REVIEW

**Comment by sethmlarson** [comment]
Gotcha, so the problem really is that we're swallowing the `RecursionError` out of `except Exception:`? If that's the case let's do something simpler, instead the very first branch under `except Exception:` check for `RecursionError` and reraise without modifying the exception.

**Comment by hmilkovi** [comment]
Yes that's the case :) . Does accepting CLA take some time to propagate or as I have signed it?

**Comment by sethmlarson** [comment]
@hmilkovi Did you sign the CLA with the same email address as the one used in your Git commits?

File: elasticsearch/_async/http_aiohttp.py [hunk 1]
```diff
@@ -303,7 +303,8 @@ async def perform_request(
         # We want to reraise a cancellation.
         except asyncio.CancelledError:
             raise
-
+        except RecursionError:
```
Comment (sethmlarson) [inline elasticsearch/_async/http_aiohttp.py:306]: Let's use the tuple here ie:

```python
except (asyncio.CancelledError, RecursionError):
    raise
```

Comment (hmilkovi) [reply elasticsearch/_async/http_aiohttp.py]: Yes are right, done.

**Comment by hmilkovi** [comment]
CLA check (read up in docs that I need to comment again)

**Comment by hmilkovi** [comment]
I have changed email on my last commit. Do I need to change it on all?
I want to contribute as private entity with private email.

Sorry for bothering you with that.

**Comment by sethmlarson** [comment]
Yeah creating a new PR is probably best

**Comment by hmilkovi** [comment]
Closing this one for https://github.com/elastic/elasticsearch-py/pull/1609 .
Now CLA is fine :)

**Pull request closed without merging**